broker.go

```diff
@@ -1570,7 +1570,7 @@ func (b *Broker) createSaslAuthenticateRequest(msg []byte) *SaslAuthenticateRequ
 func buildClientFirstMessage(token *AccessToken) ([]byte, error) {
 	var ext string
 
-	if token.Extensions != nil && len(token.Extensions) > 0 {
+	if token != nil && token.Extensions != nil && len(token.Extensions) > 0 {
```

Comment:
Actually, line 1580 dereferences `token` as well.

We should probably instead add a `if token == nil { return a short-circuit literal }` to avoid all possible derefs.